### PR TITLE
fix(modal): [modal] after the window size is changed, the window is displayed in the center

### PR DESCRIPTION
**Pull request by James-9696**
…nter

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `footer-dragable` property for the modal, allowing users to drag the footer for improved layout flexibility.
	- Added a `resetModalViewPosition` function to dynamically adjust the modal's position based on viewport size, enhancing responsiveness.

- **Bug Fixes**
	- Improved modal positioning logic to ensure it remains centered and respects margin sizes during window resizing.
	
- **Chores**
	- Updated the version of the `@opentiny/vue-renderless` package from `3.18.1` to `3.18.2`.

### CODE REVIEW

File: packages/renderless/src/modal/index.ts [hunk 1]
```diff
@@ -294,13 +294,13 @@ export const open =
...
               modalBoxElem.style.left = `${clientVisibleWidth / 2 - modalBoxElem.offsetWidth / 2}px`
             }
-
             if (
               modalBoxElem.offsetHeight + modalBoxElem.offsetTop + (props.marginSize as number) >
               clientVisibleHeight
             ) {
               modalBoxElem.style.top = `${props.marginSize}px`
             }
+            on(window, 'resize', api.resetModalViewPosition)
```
Comment (zzcr) [inline packages/renderless/src/modal/index.ts:304]: 组件卸载的时候需要移除对应的事件

Comment (James-9696) [reply packages/renderless/src/modal/index.ts:304]: 已修改

File: packages/renderless/src/modal/index.ts [hunk 2]
```diff
@@ -919,3 +920,11 @@ export const showScrollbar = (lockScrollClass) => () => {
 export const hideScrollbar = (lockScrollClass) => () => {
   removeClass(document.body, lockScrollClass)
 }
+
+export const resetModalViewPosition = (api: IModalApi) => () => {
+  const modalBoxElement = api.getBox()
+  const viewportWindow = getViewportWindow()
+  let clientVisibleWidth =
```
Comment (chenxi-20) [inline packages/renderless/src/modal/index.ts]: 这个let改为const应该是比较合适的~

Comment (James-9696) [reply packages/renderless/src/modal/index.ts]: 已修改

File: packages/renderless/src/modal/vue.ts [hunk 3]
```diff
@@ -120,6 +122,7 @@ export const renderless = (
     mousedownEvent: mousedownEvent({ api, nextTick, props, state, emit, isMobileFirstMode }),
     dragEvent: dragEvent({ api, emit, parent, props, state }),
     resetDragStyle: resetDragStyle(api),
+    resetModalViewPosition: resetModalViewPosition(api),
```
Comment (zzcr) [inline packages/renderless/src/modal/vue.ts:125]: 如果需要发补丁版本，可以帮忙升级下renderless的版本号哈

Comment (James-9696) [reply packages/renderless/src/modal/vue.ts:125]: 已修改，需要发个补丁版本的给用户

**Review by zzcr — APPROVED** [review]

**Pull request merged**